### ISSUE TITLE
Check calculation of interior area

### ISSUE THREAD

**Issue by PRemmen** [RWTH-EBC/TEASER#474]
## What is the problem?

- For some archetype buildings (especially residentail with one floor) the calculated interior (floor+wall+ceiling) seems to be very high ( e.g. 1 Floor , net leased area around 150 results in almost 400 m² interior walls).

## Why do we want to solve it?

- especially for residential buildings the inner walls do have an effect on the total heating demand 

## How do we want to solve it?

- check if the calculation works as expected
- after that: write unit tests

**Comment by mla-srs** [comment]
The calculation of the interior area is calculated (in the thermalzone) as followed:        


       for floor in self.floors:
            floor.area = (
                (  self.parent.number_of_floors - 1) /
                self.parent.number_of_floors) * self.area
        for ceiling in self.ceilings:
            ceiling.area = (
                (self.parent.number_of_floors - 1) /
                self.parent.number_of_floors) * self.area

        for wall in self.inner_walls:
            typical_area = self.typical_length * self.typical_width

            avg_room_nr = self.area / typical_area

            wall.area = (avg_room_nr * (self.typical_length *
                                        self.parent.height_of_floors +
                                        2 * self.typical_width *
                                        self.parent.height_of_floors))

with typical_length=3 and typical_width=6 for usage=living in UseCondition.xml

**Comment by mla-srs** [comment]
Does anyone have an idea where I can find the solution I can compare this to? 

Until now I looked into the teaser documentation and into IWUs Paper (
Projekt: „Entwicklung eines vereinfachten,
statistisch abgesicherten Verfahrens zur
Erhebung von Gebäudedaten für die Erstellung
des Energieprofils von Gebäuden“
Kurztitel: „Kurzverfahren Energieprofil“).

But couldn't find any calculation for the interior walls.

**Comment by PRemmen** [comment]
@mlauster may help your here @mla-srs 

What we could do in a first step to check the values if they make sense at all

**Comment by mlauster** [comment]
@mla-srs, @PRemmen, sorry for missing this issue. The calulation are based on DIN 277-3 (I assume this only contains definitions of terms), SIA 2024 and the paper from Hillebrand for `Retrofit Matrix` (Evaluation Tool and Retrofit Matrix for Office Buildings. In: EON Energy Research Series 4 (2012). Let me know if I should provide a copy of this.
But according to your example, @premmen, with one floor (so no ceilings or floors), given three rooms leads to a height of floors of 14.8 m. Even seven rooms (20 m² each) leads to a height of floors of 6.3 m. Am I getting something wrong?

**Comment by mlauster** [comment]
Of course, what is not considered is, that not all rooms in the house have three interior walls. In particular for small houses, there might be more rooms with less than three interior walls, since they contain a corner of exterior walls or span an entire side of the house. In such cases, the estimation might be too high. This will mainly affect single family dwellings (SFD). We could distinguish the building types in the calculation and consider only two or one and a half interior walls for (SFD). Does that sound like a reasonable approach for you?

**Comment by PGorzalka** [comment]
How about this approach? Given that it is quite straightforward, I wonder if there is any reason it was not used before:
1. calculate number of rooms (as before)
2. **total** wall area `                wall_area = (avg_room_nr
                             * (2 * self.use_conditions.typical_length
                                * height_of_floors
                                + 2 * self.use_conditions.typical_width
                                * height_of_floors))`
3. substract all outer walls, windows and vertical roof parts from the result to get interior wall area
